packages/core/src/api/events/initialize.ts

```diff
@@ -0,0 +1,37 @@
+import type { Config } from '../../config';
+import type { InternalLogger } from '../../internalLogger';
+import type { Metas } from '../../metas';
+import { TransportItem, TransportItemType, Transports } from '../../transports';
+import { getCurrentTimestamp } from '../../utils';
+import type { TracesAPI } from '../traces';
+import type { EventEvent, EventsAPI } from './types';
+
+export function initializeEventsAPI(
+  internalLogger: InternalLogger,
+  config: Config,
+  transports: Transports,
+  metas: Metas,
+  tracesApi: TracesAPI
+): EventsAPI {
+  const pushEvent: EventsAPI['pushEvent'] = (name, attributes, domain) => {
+    const item: TransportItem<EventEvent> = {
```

Comment:
Worth wrapping this in a trycatch like we do with `pushLog`? Not sure if we expect transports.execute to throw.

Reply:
Sure, won't hurt